src/nc/core/likec/TreePrinter.cpp

```diff
@@ -276,7 +276,14 @@ void TreePrinter::printSignature(const FunctionDeclaration *node) {
 }
 
 void TreePrinter::doPrint(const MemberDeclaration *node) {
-    out_ << *node->type() << ' ' << node->identifier() << ';';
+    if (!node->type()->isScalar()) {
```

Comment:
Could you please make use of nc casts?
Something like:
```
if (auto pointerType = node->type()->as<PointerType>()) {
    if (auto arrayType = pointerType->as<ArrayType>()) {
        // Print array declaration.
        return;
    }
}
// Print the usual declaration.
```
You will not need `reinterpret_cast` then.

Also, isScalar is false not only for array types. It is also false for void and structs, for example. So, it is not very correct to check for array type using isScalar().